File: Itinere_3/Spring-Boot/owner_dog/src/main/java/com/example/owner_dog/controllers/OwnerController.java
package com.example.owner_dog.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

import com.example.owner_dog.models.Owner;
import com.example.owner_dog.repositories.OwnerRepository;

@Controller
public class OwnerController {

    final OwnerRepository repo;

    public OwnerController(OwnerRepository repo) {
        this.repo = repo;
    }

    @GetMapping("/")
    String welcome() {
        return "welcome";
    }

    @GetMapping("/owners")
    String index(Model model) {
        model.addAttribute("owners", repo.findAll());
        return "/owners/index";
    }

    @GetMapping("/owners/create")
    String create() {
        return "/owners/create";
    }

    @PostMapping("/owners/store")
    String store(Owner o) {
        repo.save(o);
        return "redirect:/owners";
    }

    @GetMapping("/owners/show")
    String show(Model model, Long id) {
        model.addAttribute("o", repo.findById(id).orElse(null));
        return "/owners/show";
    }

    @GetMapping("/owners/edit")
    String edit(Model model, Long id) {
        model.addAttribute("o", repo.findById(id).orElse(null));
        return "/owners/edit";
    }

    @PostMapping("/owners/update")
    String update(Owner o) {
        repo.save(o);
        return "redirect:/owners";
    }

    @PostMapping("/owners/destroy")
    String destroy(Long id) {
        repo.deleteById(id);
        return "redirect:/owners";
    }
}

File: Itinere_3/Laravel/owner_dog/app/Http/Controllers/OwnerController.php
<?php

namespace App\Http\Controllers;

use App\Models\Owner;
use Illuminate\Http\Request;

class OwnerController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view("owners.index", ["owners" => Owner::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("owners.create");
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Owner::create($request->all());
        return redirect()->route("owners.index");
    }

    /**
     * Display the specified resource.
     */
    public function show(Owner $owner)
    {
        return view("owners.show", ["o" => $owner]);
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Owner $owner)
    {
        return view("owners.edit", ["o" => $owner]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Owner $owner)
    {
        $owner->update($request->all());
        return redirect()->route("owners.index");
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Owner $owner)
    {
        $owner->delete();
        return redirect()->route("owners.index");
    }
}
